Sources/ApodiniHTTP/ClosureBuilders/HTTP1Compatibility/ServiceSideStreaming.swift

```diff
@@ -9,24 +9,70 @@
 import Foundation
 import Apodini
 import ApodiniExtension
-import ApodiniVaporSupport
-import Vapor
+import ApodiniNetworking
+
+
+extension AsyncSequence {
+    /// Returns an `EventLoopFuture` which will fulfill with the first element in the sequence, and also calls the specified closure once with every element in the sequence
+    func firstFutureAndForEach(on eventLoop: EventLoop, objectsHandler: @escaping (Element) -> Void) -> EventLoopFuture<Element?> {
+        let promise = eventLoop.makePromise(of: Element?.self)
+        Task {
+            var idx = 0
+            for try await element in self {
+                if idx == 0 {
+                    promise.succeed(element)
+                }
+                idx += 1
+                objectsHandler(element)
+            }
+            if idx == 0 {
+                promise.succeed(nil)
+            }
+        }
+        return promise.futureResult
+    }
+}
+
 
 extension Exporter {
-    // MARK: Service Streaming Closure
+    func buildServiceSideStreamingClosure<H: Handler>(
+        for endpoint: Endpoint<H>,
+        using defaultValues: DefaultValueStore
+    ) -> (HTTPRequest) throws -> EventLoopFuture<HTTPResponse> {
+        let encoder = configuration.encoder
+        return _buildServiceSideStreamingClosure(for: endpoint, using: defaultValues) { response -> Data? in
+            if let response = response {
+                return try encoder.encode(response)
+            } else {
+                return nil
+            }
+        }
+    }
+    
     
     func buildServiceSideStreamingClosure<H: Handler>(
         for endpoint: Endpoint<H>,
-        using defaultValues: DefaultValueStore) -> (Vapor.Request) throws -> EventLoopFuture<Vapor.Response> {
+        using defaultValues: DefaultValueStore
+    ) -> (HTTPRequest) throws -> EventLoopFuture<HTTPResponse> where H.Response.Content == Blob {
+        _buildServiceSideStreamingClosure(for: endpoint, using: defaultValues) { (response: Blob?) -> Data? in
+            precondition(response?.byteBuffer.readerIndex == 0)
+            return response?.byteBuffer.getAllData()
+        }
+    }
+    
+    
+    private func _buildServiceSideStreamingClosure<H: Handler>(
```

Comment:
I'm also not really knowledgeable on the core of Apodinis Exporter API and async stuff. But, to my limited knowledge, this looks fine. Not sure if this helps at all 😅